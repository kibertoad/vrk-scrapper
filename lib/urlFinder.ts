// const urlRegex = /\'?srcUrl=\'(.*)\'.html\'/

export function findAllUrls(text: string): readonly string[] {
  const result = text.match(/\?srcUrl(.*)\.html/gi) ?? []
  return result.map((srcUrl) => {
    return srcUrl.replace('?srcUrl=', '')
  })
}
