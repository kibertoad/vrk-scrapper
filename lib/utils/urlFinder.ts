export function findAllUrls(text: string): readonly string[] {
  const result = text.match(/\?srcUrl(.*)\.html/gi) ?? []
  return result
}
