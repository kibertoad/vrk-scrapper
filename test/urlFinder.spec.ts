import { findAllUrls } from '../lib/utils/urlFinder'
import fs from 'fs'
import path from 'path'

const tableHTML = fs
  .readFileSync(path.resolve(__dirname, 'resources/single_mandate_table.html'))
  .toString()

describe('urlFinder', () => {
  it('correctly finds urls', () => {
    const result = findAllUrls(tableHTML)
    expect(result).toMatchSnapshot()
  })
})
