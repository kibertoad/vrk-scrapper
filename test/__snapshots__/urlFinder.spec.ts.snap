// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`urlFinder correctly finds urls 1`] = `
Array [
  "/rinkimai/1104/1/1746/rezultatai/lt/rezultataiVienmRpg_rpgId-20722.html",
  "/rinkimai/1104/1/1746/rezultatai/lt/rezultataiVienmRpg_rpgId-20940.html",
  "/rinkimai/1104/1/1746/rezultatai/lt/rezultataiVienmRpg_rpgId-20732.html",
  "/rinkimai/1104/1/1746/rezultatai/lt/rezultataiVienmRpg_rpgId-20898.html",
  "/rinkimai/1104/1/1746/rezultatai/lt/rezultataiVienmRpg_rpgId-20740.html",
  "/rinkimai/1104/1/1746/rezultatai/lt/rezultataiVienmRpg_rpgId-20900.html",
  "/rinkimai/1104/1/1746/rezultatai/lt/rezultataiVienmRpg_rpgId-20902.html",
  "/rinkimai/1104/1/1746/rezultatai/lt/rezultataiVienmRpg_rpgId-20904.html",
  "/rinkimai/1104/1/1746/rezultatai/lt/rezultataiVienmRpg_rpgId-20906.html",
  "/rinkimai/1104/1/1746/rezultatai/lt/rezultataiVienmRpg_rpgId-20908.html",
  "/rinkimai/1104/1/1746/rezultatai/lt/rezultataiVienmRpg_rpgId-20690.html",
  "/rinkimai/1104/1/1746/rezultatai/lt/rezultataiVienmRpg_rpgId-20688.html",
  "/rinkimai/1104/1/1746/rezultatai/lt/rezultataiVienmRpg_rpgId-20912.html",
  "/rinkimai/1104/1/1746/rezultatai/lt/rezultataiVienmRpg_rpgId-20742.html",
  "/rinkimai/1104/1/1746/rezultatai/lt/rezultataiVienmRpg_rpgId-20938.html",
  "/rinkimai/1104/1/1746/rezultatai/lt/rezultataiVienmRpg_rpgId-20914.html",
  "/rinkimai/1104/1/1746/rezultatai/lt/rezultataiVienmRpg_rpgId-20744.html",
  "/rinkimai/1104/1/1746/rezultatai/lt/rezultataiVienmRpg_rpgId-20916.html",
  "/rinkimai/1104/1/1746/rezultatai/lt/rezultataiVienmRpg_rpgId-20910.html",
  "/rinkimai/1104/1/1746/rezultatai/lt/rezultataiVienmRpg_rpgId-20944.html",
  "/rinkimai/1104/1/1746/rezultatai/lt/rezultataiVienmRpg_rpgId-20918.html",
  "/rinkimai/1104/1/1746/rezultatai/lt/rezultataiVienmRpg_rpgId-20920.html",
  "/rinkimai/1104/1/1746/rezultatai/lt/rezultataiVienmRpg_rpgId-20728.html",
  "/rinkimai/1104/1/1746/rezultatai/lt/rezultataiVienmRpg_rpgId-20924.html",
  "/rinkimai/1104/1/1746/rezultatai/lt/rezultataiVienmRpg_rpgId-20922.html",
  "/rinkimai/1104/1/1746/rezultatai/lt/rezultataiVienmRpg_rpgId-20926.html",
  "/rinkimai/1104/1/1746/rezultatai/lt/rezultataiVienmRpg_rpgId-20738.html",
  "/rinkimai/1104/1/1746/rezultatai/lt/rezultataiVienmRpg_rpgId-20928.html",
  "/rinkimai/1104/1/1746/rezultatai/lt/rezultataiVienmRpg_rpgId-20720.html",
  "/rinkimai/1104/1/1746/rezultatai/lt/rezultataiVienmRpg_rpgId-20930.html",
  "/rinkimai/1104/1/1746/rezultatai/lt/rezultataiVienmRpg_rpgId-20932.html",
  "/rinkimai/1104/1/1746/rezultatai/lt/rezultataiVienmRpg_rpgId-20934.html",
  "/rinkimai/1104/1/1746/rezultatai/lt/rezultataiVienmRpg_rpgId-20936.html",
  "/rinkimai/1104/1/1746/rezultatai/lt/rezultataiVienmRpg_rpgId-20730.html",
  "/rinkimai/1104/1/1746/rezultatai/lt/rezultataiVienmRpg_rpgId-20708.html",
  "/rinkimai/1104/1/1746/rezultatai/lt/rezultataiVienmRpg_rpgId-20750.html",
  "/rinkimai/1104/1/1746/rezultatai/lt/rezultataiVienmRpg_rpgId-20752.html",
  "/rinkimai/1104/1/1746/rezultatai/lt/rezultataiVienmRpg_rpgId-20754.html",
  "/rinkimai/1104/1/1746/rezultatai/lt/rezultataiVienmRpg_rpgId-20692.html",
  "/rinkimai/1104/1/1746/rezultatai/lt/rezultataiVienmRpg_rpgId-20756.html",
  "/rinkimai/1104/1/1746/rezultatai/lt/rezultataiVienmRpg_rpgId-20758.html",
  "/rinkimai/1104/1/1746/rezultatai/lt/rezultataiVienmRpg_rpgId-20946.html",
  "/rinkimai/1104/1/1746/rezultatai/lt/rezultataiVienmRpg_rpgId-20760.html",
  "/rinkimai/1104/1/1746/rezultatai/lt/rezultataiVienmRpg_rpgId-20762.html",
  "/rinkimai/1104/1/1746/rezultatai/lt/rezultataiVienmRpg_rpgId-20764.html",
  "/rinkimai/1104/1/1746/rezultatai/lt/rezultataiVienmRpg_rpgId-20694.html",
  "/rinkimai/1104/1/1746/rezultatai/lt/rezultataiVienmRpg_rpgId-20948.html",
  "/rinkimai/1104/1/1746/rezultatai/lt/rezultataiVienmRpg_rpgId-20766.html",
  "/rinkimai/1104/1/1746/rezultatai/lt/rezultataiVienmRpg_rpgId-20768.html",
  "/rinkimai/1104/1/1746/rezultatai/lt/rezultataiVienmRpg_rpgId-20770.html",
  "/rinkimai/1104/1/1746/rezultatai/lt/rezultataiVienmRpg_rpgId-20696.html",
  "/rinkimai/1104/1/1746/rezultatai/lt/rezultataiVienmRpg_rpgId-20772.html",
  "/rinkimai/1104/1/1746/rezultatai/lt/rezultataiVienmRpg_rpgId-20746.html",
  "/rinkimai/1104/1/1746/rezultatai/lt/rezultataiVienmRpg_rpgId-20942.html",
  "/rinkimai/1104/1/1746/rezultatai/lt/rezultataiVienmRpg_rpgId-20950.html",
  "/rinkimai/1104/1/1746/rezultatai/lt/rezultataiVienmRpg_rpgId-20700.html",
  "/rinkimai/1104/1/1746/rezultatai/lt/rezultataiVienmRpg_rpgId-20774.html",
  "/rinkimai/1104/1/1746/rezultatai/lt/rezultataiVienmRpg_rpgId-20698.html",
  "/rinkimai/1104/1/1746/rezultatai/lt/rezultataiVienmRpg_rpgId-20776.html",
  "/rinkimai/1104/1/1746/rezultatai/lt/rezultataiVienmRpg_rpgId-20702.html",
  "/rinkimai/1104/1/1746/rezultatai/lt/rezultataiVienmRpg_rpgId-20778.html",
  "/rinkimai/1104/1/1746/rezultatai/lt/rezultataiVienmRpg_rpgId-20780.html",
  "/rinkimai/1104/1/1746/rezultatai/lt/rezultataiVienmRpg_rpgId-20712.html",
  "/rinkimai/1104/1/1746/rezultatai/lt/rezultataiVienmRpg_rpgId-20710.html",
  "/rinkimai/1104/1/1746/rezultatai/lt/rezultataiVienmRpg_rpgId-20706.html",
  "/rinkimai/1104/1/1746/rezultatai/lt/rezultataiVienmRpg_rpgId-20748.html",
  "/rinkimai/1104/1/1746/rezultatai/lt/rezultataiVienmRpg_rpgId-20782.html",
  "/rinkimai/1104/1/1746/rezultatai/lt/rezultataiVienmRpg_rpgId-20784.html",
  "/rinkimai/1104/1/1746/rezultatai/lt/rezultataiVienmRpg_rpgId-20704.html",
  "/rinkimai/1104/1/1746/rezultatai/lt/rezultataiVienmRpg_rpgId-20786.html",
  "/rinkimai/1104/1/1746/rezultatai/lt/rezultataiVienmRpg_rpgId-20734.html",
]
`;
